web/src/models/DAG.model.ts

```diff
@@ -13,6 +13,7 @@ function getNodesDatumFromSpans(spans: Span[], type: NodeTypesEnum): INodeDatum<
 }
 
 function DAG(spans: Span[], type: NodeTypesEnum) {
+  // TODO: this runs twice for the list of spans
```

Comment:
We might be able to use the tree data structure of the trace here, instead of sorting, by running a recursive `flatMap` instead